src/encoder.rs

```diff
@@ -324,6 +324,16 @@ impl<W: Write> Encoder<W> {
         writer.write_le(0u8) // aspect ratio
     }
 
+    /// Gets a reference to the writer instance used by this encoder.
+    pub fn get_ref(&self) -> &W {
+        self.w.as_ref().unwrap()
+    }
+
+    /// Gets a mutable reference to the writer instance used by this encoder.
+    pub fn get_mut(&mut self) -> &mut W {
+        self.w.as_mut().unwrap()
+    }
```

Comment:
The standard library `BufWriter` includes this piece of advice:

> It is inadvisable to directly write to the underlying writer.

This is even more clearly the case for our method, as doing so _could_ corrupt the gif. Can you add a similar warning?